statsmodels/tsa/ar_model.py

```diff
@@ -478,6 +476,9 @@ def _wrap_prediction(self, prediction, start, end):
             else:
                 index = pd.RangeIndex(end)
         index = index[start:end]
+        prediction = prediction[-len(index):]
+        if not isinstance(self.data.orig_endog, (pd.Series, pd.DataFrame)):
```

Comment:
You need to early exit in case of a NumPy array in the first line.  Is the function broken for numpy array inputs of just Series/DataFrame?

Reply:
Yes, I realised that too late, didn't get the tests to run locally, not sure though if that should happen in this function or wherever the predictions are generated ...